### PR TITLE
fix: Less error logging, more retrying in ddata rememer entities

### PR DESCRIPTION
**Pull request by johanandren**
References #31801

 * DDataRememberEntitiesShardStore: Retry a couple of times if reading the shards times out/fails and only log at warning until finally giving up
 * DDataRememberEntitiesCoordinatorStore: Already was retrying indefinitely, so only log at warning

### CODE REVIEW

**Review by patriknw — COMMENTED** [review]
looking good

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesShardStore.scala [hunk 1]
```diff
@@ -91,6 +91,8 @@ private[akka] final class DDataRememberEntitiesShardStore(
   // Note that the timeout is actually updatingStateTimeout / 4 so that we fit 3 retries and a response in the timeout before the shard sees it as a failure
   private val writeMajority = WriteMajority(settings.tuningParameters.updatingStateTimeout / 4, majorityMinCap)
```
[the remark this answers is not in the document]
Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesShardStore.scala]: Done in  [e186e41](https://github.com/akka/akka/pull/31802/commits/e186e4125d5a523d64cc434a2ad2527f1d57f7ab)

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesCoordinatorStore.scala [hunk 2]
```diff
@@ -74,7 +74,7 @@ private[akka] final class DDataRememberEntitiesCoordinatorStore(
       onGotAllShards(Set.empty)
 
     case Replicator.GetFailure(AllShardsKey, _) =>
-      log.error(
+      log.warning(
         "The ShardCoordinator was unable to get all shards state within 'waiting-for-state-timeout': {} millis (retrying)",
```
[the remark this answers is not in the document]
Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesCoordinatorStore.scala]: Changed in [635607d](https://github.com/akka/akka/pull/31802/commits/635607d0177c4dccb272c37fb9e8029fb70409cf) also changing to more clear logs for both store actors.

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesCoordinatorStore.scala [hunk 3]
```diff
@@ -46,6 +46,7 @@ private[akka] final class DDataRememberEntitiesCoordinatorStore(
   private val writeMajority = WriteMajority(settings.tuningParameters.updatingStateTimeout, majorityMinCap)
```
Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesCoordinatorStore.scala:47]: these should also be changed to Plus, so we use the same approach for all of them

Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesCoordinatorStore.scala:55]: https://github.com/akka/akka/pull/31802/commits/7db9a7a777a4dc5a1d6fb9baf350b561e88aea41

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**